nightfall-client/src/utils/getCommitmentInfo.mjs

```diff
@@ -42,14 +45,36 @@ export const getCommitmentInfo = async txInfo => {
   const value = totalValueToSend + addedFee;
   const feeValue = fee.bigInt - addedFee;
 
-  const commitments = await findUsableCommitmentsMutex(
-    compressedZkpPublicKey,
-    ercAddress,
-    tokenId,
-    maticAddress,
-    value,
-    feeValue,
-  );
+  logger.debug(`using user provided commitments: ${providedCommitments !== undefined}`);
+
+  let commitments;
+  if (providedCommitments) {
+    // look up the hashes
+    const commitmentHashes = providedCommitments.map(c => c.toString());
+    logger.debug({ msg: 'looking up these commitment hashes:', commitmentHashes });
+    const rawCommitments = await getCommitmentsByHash(commitmentHashes);
```

Comment:
I think there are several scenarios to be considered here: 
- What happens if a user provides a fake commitment hash?
- What happens if a user provides a commitment for which he isn't the owner? Or a commitment whose preimage is a different ercAddress? 
- What happens if a commitment is already nullified or pending nullification?
- What happens if the total sum of the commitments isn't enough for what the user is intending to transfer/withdraw?

Hint: Take a look at `verifyEnoughCommitments` function in the commitment storage